cmd/specdiff/differ_test.go

```diff
@@ -0,0 +1,282 @@
+package specdiff
+
+import (
+	"io/ioutil"
+	"reflect"
+	"testing"
+
+	"github.com/apigee/registry/rpc"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/tufin/oasdiff/diff"
+	"google.golang.org/protobuf/testing/protocmp"
+)
+
+type testStruct struct {
+	Name    string            `json:"name,omitempty"`
+	TestMap map[string]string `json:"testmap,omitempty"`
+}
+
+func TestDiffProtoStruct(t *testing.T) {
+	tests := []struct {
+		desc         string
+		baseSpec     string
+		revisionSpec string
+		wantProto    *rpc.Diff
+	}{
+		{
+			desc:         "Struct Diff Added",
+			baseSpec:     "./test-specs/base-test.yaml",
+			revisionSpec: "./test-specs/struct-test-add.yaml",
+			wantProto: &rpc.Diff{
+				Added: []string{
+					"components.schemas.Pet.required.age",
+					"components.schemas.Pet.properties.age",
+				},
+				Deleted:      []string{},
+				Modification: map[string]*rpc.Diff_ValueChange{},
+			},
+		},
+		{
+			desc:         "Struct Diff Deleted",
+			baseSpec:     "./test-specs/base-test.yaml",
+			revisionSpec: "./test-specs/struct-test-delete.yaml",
+			wantProto: &rpc.Diff{
+				Added: []string{},
+				Deleted: []string{
+					"components.schemas.Pet.required.name",
+				},
+				Modification: map[string]*rpc.Diff_ValueChange{},
+			},
+		},
+		{
+			desc:         "Struct Diff Modified",
+			baseSpec:     "./test-specs/base-test.yaml",
+			revisionSpec: "./test-specs/struct-test-modify.yaml",
+			wantProto: &rpc.Diff{
+				Added:   []string{},
+				Deleted: []string{},
+				Modification: map[string]*rpc.Diff_ValueChange{
+					"info.version": {
+						To:   "1.0.1",
+						From: "1.0.0",
+					},
+					"components.schemas.Pet.properties.tag.type": {
+						To:   "integer",
+						From: "string",
+					},
+					"components.schemas.Pet.properties.tag.format": {
+						To: "int64",
+					},
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			yamlFile, err := ioutil.ReadFile(test.baseSpec)
+			if err != nil {
+				t.Fatalf("Failed to get base spec yaml: %s", err)
+			}
+			yamlFile2, err := ioutil.ReadFile(test.revisionSpec)
+			if err != nil {
+				t.Fatalf("Failed to get revision spec yaml: %s", err)
+			}
+			diffProto, err := GetDiff(yamlFile, yamlFile2)
+			if err != nil {
+				t.Fatalf("Failed to get diff.Diff: %s", err)
+			}
+			opts := cmp.Options{
+				protocmp.Transform(),
+				cmpopts.SortSlices(func(a, b string) bool { return a < b }),
+			}
+			if !cmp.Equal(test.wantProto, diffProto, opts) {
+				t.Errorf("Test %+v returned unexpected diff (-want +got):\n%s", test.desc, cmp.Diff(test.wantProto, diffProto, opts))
```

Comment:
On line 76 the description is already included in the name of the subtest, so printing it again isn't necessary. It's a good idea to call out the specific function that had an unexpected return value, i.e. GetDiff returned unexpected diff...